app/cloudinit/user_data/gcp_linux

```diff
@@ -140,38 +140,89 @@ runcmd:
     
     # Bind mount directories
     echo "Setting up bind mounts..." | tee -a /var/log/nvme-setup.log
-    for dir in /tmp /var/lib/docker /home; do
+    for dir in /tmp /var/lib/docker /home /harness; do
       name=$(basename "$dir")
       target="$MOUNT_POINT/$name"
       
       echo "Processing directory: $dir" | tee -a /var/log/nvme-setup.log
       echo "Target directory: $target" | tee -a /var/log/nvme-setup.log
       
+      # Ensure source directory exists
+      if [ ! -d "$dir" ]; then
+        echo "Creating source directory $dir as it doesn't exist" | tee -a /var/log/nvme-setup.log
+        mkdir -p "$dir"
+        chmod 755 "$dir"
+      fi
+      
       # Create target directory
       echo "Creating target directory..." | tee -a /var/log/nvme-setup.log
       mkdir -p "$target"
       
       # Copy existing content
       echo "Copying content from $dir to $target..." | tee -a /var/log/nvme-setup.log
-      if [ -d "$dir" ] && [ "$(ls -A $dir 2>/dev/null)" ]; then
+      if [ "$(ls -A $dir 2>/dev/null)" ]; then
         cp -a "$dir/"* "$target/" 2>/dev/null || echo "Warning: Some files could not be copied" | tee -a /var/log/nvme-setup.log
       else
-        echo "Source directory is empty or doesn't exist" | tee -a /var/log/nvme-setup.log
+        echo "Source directory is empty" | tee -a /var/log/nvme-setup.log
       fi
       
       # Bind mount
       echo "Binding $target to $dir..." | tee -a /var/log/nvme-setup.log
+      umount "$dir" 2>/dev/null  # Unmount if already mounted
       mount --bind "$target" "$dir" | tee -a /var/log/nvme-setup.log
       if [ $? -ne 0 ]; then
         echo "Failed to bind mount $target to $dir" | tee -a /var/log/nvme-setup.log
       else
         echo "Successfully bind mounted $target to $dir" | tee -a /var/log/nvme-setup.log
       fi
+      
+      # Add to fstab for persistence
+      if ! grep -q "$target $dir" /etc/fstab; then
+        echo "$target $dir none bind 0 0" >> /etc/fstab
+        echo "Added bind mount to fstab for $dir" | tee -a /var/log/nvme-setup.log
+      fi
     done
     
     # Verify mounts and log completion
-    mount | grep -E "$DEV_PATH|$MOUNT_POINT" | tee -a /var/log/nvme-setup.log
-    echo "NVME setup complete at $(date)" | tee -a /var/log/nvme-setup.log
+    echo -e "\nVerifying all mounts:" | tee -a /var/log/nvme-setup.log
+    mount | grep -E "$DEV_PATH|$MOUNT_POINT|/tmp|/var/lib/docker|/home|/harness" | tee -a /var/log/nvme-setup.log
+    
+    # Check if all directories are properly mounted
+    echo -e "\nChecking if all directories are properly mounted:" | tee -a /var/log/nvme-setup.log
+    ALL_MOUNTED=true
```

Comment:
For checks like this you should assume false first and then validate if it's actually true IMO